scripts/code_generation/commit_info_gen/src/generate_commit_info.py

```diff
@@ -0,0 +1,56 @@
+import os
+import string
+from argparse import ArgumentParser
+import jinja2
+import git
+
+# validates data dict
+# data["hash"]: 7 char string with short commit hash
+# data["clean"]: "0" if dirty, "1" is clean
+def validateData(data):
+  return (
+    len(data["hash"]) == 7 # check length of hash
+    and all([c in string.hexdigits for c in data["hash"]])  # check hash is valid hexadecimal
+    and (data["clean"] == "0" or data["clean"] == "1") # check clean bit
```

Comment:
instead of using comments, use 3 variable names.

Reply:
Fixed!